karyon3-core/src/main/java/com/netflix/karyon/Karyon.java

```diff
@@ -333,84 +330,72 @@ public LifecycleInjector start() {
-                                bind(String[].class).annotatedWith(Arguments.class).toInstance(new String[]{});
-                            }
-                        }
-                    })
-                    .build())
-                .with(overrideModules);
+            List<Module> coreModules = new ArrayList<>();
+            coreModules.addAll(modules);
+            coreModules.add(module);
+            coreModules.add(new CoreModule());
+            coreModules.add(new AbstractModule() {
+                @Override
+                protected void configure() {
+                    bind(KaryonFeatureSet.class).toInstance(featureSet);
+                    bind(LifecycleManager.class).toInstance(manager);
+                    bind(new TypeLiteral<Set<String>>() {}).annotatedWith(Profiles.class).toInstance(profiles);
+                    bind(String[].class).annotatedWith(Arguments.class).toInstance(args);
+                }
+            });
             
-            for (Element binding : Elements.getElements(coreModule)) {
-                LOG.debug("Binding : {}", binding);
+            for (ModuleListTransformer transformer : transformers) {
+                coreModules = transformer.transform(Collections.unmodifiableList(coreModules));
             }
-            
-            Injector injector = Guice.createInjector(
-                stage,
-                coreModule,
-                new AbstractModule() {
-                    @Override
-                    protected void configure() {
-                        for (Key<?> key : ElementsEx.getAllUnboundKeys(Elements.getElements(coreModule))) {
-                            for (MatchingAutoBinder factory : Karyon.this.autoBinders) {
-                                if (factory.configure(binder(), key)) {
-                                    break;
-                                }
-                            }
+
+            // Loop through all elements that have been bound and look for any missing bindings.  
+            // For missing bindings try to call the registered AutoBinders.  
+            // Since an AutoBinder may add additional bindings the processes is repeated until 
+            // no new auto bindings have been created
+            //
+            // TODO: This is probably not the most efficient way of doing this.  This code could be 
+            //       optimized by storing state and only evaluating newly added modules for bindings.
+            List<AutoBinder> prioritizedAutoBinders = autoBinders.stream().sorted(byPriority).collect(Collectors.toList());
+            boolean done = false;
+            while (!done) {
+                done = true;
+                for (Key<?> key : ElementsEx.getAllUnboundKeys(Elements.getElements(coreModules))) {
+                    for (AutoBinder factory : prioritizedAutoBinders) {
+                        if (factory.matches(key)) {
+                            coreModules.add(factory.getModuleForKey(key));
```

Comment:
When do conflicts blow up?  Presumably below when Guice.createInjector is called.  Is there any way to make it clear the issue is that there are multiple AutoBinder binding the same type, or is that going to be nearly impossible to debug?